### PR TITLE
Fix: Remove hardcoded sampling-strategies.json dependency in Jaeger in-one binary

### LINKED ISSUE #6338: [bug]: v2 all-in-one binary cannot find sampling strategies file
### Requirement

As a developer using jeager on my machine,

I would like a convenient non-docker all-in-one build for 2.x like we have in 1.x

### Problem

Being able to quickly start using jeager without requiring Docker 

### Proposal

A non-docker all-in-one build for 2.x like we have in 1.x

### Workaround

Save the content of [sampling-strategies.json](https://github.com/jaegertracing/jaeger/blob/main/cmd/jaeger/sampling-strategies.json) in a new folder cmd/jaeger created next to jaeger.exe

### PR DESCRIPTION
**Pull request by yunmaoQu**
## Which problem is this PR solving?
- fix #6338

## Description of the changes
-  Removed the hardcoded path in the default configuration (`all-in-one.yaml`).
- Updated the code to log a warning if the file is missing and proceed with default behavior.
- Deleted the unused `sampling-strategies.json` file.

## How was this change tested?
- passed

## Checklist
- [ ] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [ ] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [ ] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `npm run lint` and `npm run test`

### CODE REVIEW

File: cmd/jaeger/main.go [hunk 1]
```diff
@@ -24,6 +25,14 @@ func main() {
 		command,
 	)
 
+	if err := command.Execute(); err != nil {
+		log.Fatal(err)
+	}
+	strategiesFile := "./cmd/jaeger/sampling-strategies.json"
```
Comment (yurishkuro) [inline cmd/jaeger/main.go:31]: this logic does not belong in main

File: cmd/jaeger/internal/all-in-one.yaml [hunk 2]
```diff
@@ -28,12 +28,9 @@ extensions:
           max_traces: 100000
 
   remote_sampling:
-    # We can either use file or adaptive sampling strategy in remote_sampling
-    file:
-      path: ./cmd/jaeger/sampling-strategies.json
```
Comment (yurishkuro) [inline cmd/jaeger/internal/all-in-one.yaml:33]: We still want to use a file based strategy by default. A user running all-in-one should be able to override the path via `--set` flag.

**Review by yurishkuro — CHANGES_REQUESTED** [review]

**Comment by avinpy-255** [comment]
Hi @yurishkuro 
I tried running the Jaeger binary locally and read the error message (same as in the GitHub comments). I think the issue is that it cannot find the sampling-strategies.json file. I believe defining the file path for the sampling strategies in the config file might resolve the issue, but I'm not sure.

I also checked the config files under cmd/jaeger/internal/config.yaml and found this line, which is commented out:

```
remote_sampling:      
# You can either use file or adaptive sampling strategy in remote_sampling      
# file:      
#   path: ./cmd/jaeger/sampling-strategies.json   
```

I'm wondering why this is commented out. Is this the file where I should make changes?

**Comment by yurishkuro** [comment]
It's not commented out https://github.com/jaegertracing/jaeger/blob/4ecb0860ce7ae901f8d77261b4022e5030288073/cmd/jaeger/internal/all-in-one.yaml#L30-L34

But it is incorrect to leave it like this because it assumes some random location of the config. My proposal would be to keep the `file:` strategy selected, but comment out the file name. This will fail the validatin so we'd need to fix that first, then in the actual strategy implementation it should allow empty file name and return a default strategy (probabilistic with p=1.0), because that would be equivalent to our [existing config](https://github.com/jaegertracing/jaeger/blob/main/cmd/jaeger/sampling-strategies.json) but without the bogus service names.

**Comment by yurishkuro** [comment]
Closing this in favor of #6431

**Pull request closed without merging**